sdk/resourcemanager/azure-resourcemanager-resources/src/test/java/com/azure/resourcemanager/resources/FeaturesTests.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.resources;
+
+import com.azure.resourcemanager.resources.models.Feature;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FeaturesTests extends ResourceManagementTest {
+
+    @Test
+    public void canListAndRegisterFeature() {
+        List<Feature> features = resourceClient.features().list().stream().collect(Collectors.toList());
+        Assertions.assertNotNull(features);
+
+        features.stream()
+            .filter(f -> "NotRegistered".equals(f.state()))
+            .findFirst()
+            .ifPresent(feature -> resourceClient.features()
+                .register(feature.resourceProviderName(), feature.featureName()));
```

Comment:
I think find a `Registered` one is better (i think it could be registered). Otherwise when all providers are registered, the test can not test anything.

Reply:
For now most feature is NotRegistered (and we got lots of them).

BTW, if I find Registered, I cannot call unregister on it as someone may depends on that feature.